File: rb/features.rb
# TelegramChannel SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TelegramChannelFeatures
  def self.make_feature(name)
    case name
    when "base"
      TelegramChannelBaseFeature.new
    when "test"
      TelegramChannelTestFeature.new
    else
      TelegramChannelBaseFeature.new
    end
  end
end

File: py/features.py
# TelegramChannel SDK feature factory

from feature.base_feature import TelegramChannelBaseFeature
from feature.test_feature import TelegramChannelTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TelegramChannelBaseFeature(),
        "test": lambda: TelegramChannelTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
